File: front/src/api/job.ts
import { request } from '@/utils/request'
import type { ApiResponse, PageResponse } from '@/types/api'

// Job查询参数
export interface JobQueryDTO {
  clusterId: string
  namespace?: string
  name?: string
  page?: number
  pageSize?: number
  keyword?: string
}

// Job视图对象
export interface JobVO {
  name: string
  namespace: string
  parallelism?: number
  completions?: number
  activeDeadlineSeconds?: number
  backoffLimit?: number
  ttlSecondsAfterFinished?: number
  active: number
  succeeded: number
  failed: number
  startTime?: string
  completionTime?: string
  generation: number
  version: string
  uid: string
  resourceVersion: string
  annotations?: Record<string, string>
  labels?: Record<string, string>
  creationTimestamp: string
  createTime: string
  yaml?: string
}

// Job API
export const jobApi = {
  // 获取Job列表
  getJobs: (params: JobQueryDTO): Promise<ApiResponse<PageResponse<JobVO>>> => {
    return request.get('/job/list', { params })
  }
}

File: internal/web/vo/job.go
package vo

import (
	batchv1 "k8s.io/api/batch/v1"
	"time"
)

type JobVO struct {
	BaseVO
	Alias       string    `json:"alias"`
	Describe    string    `json:"describe"`
	CreateTime  time.Time `json:"createTime"`
	LastRunTime time.Time `json:"lastRunTime"`
	Yaml        string    `json:"yaml"`
}

func Job2VO(job batchv1.Job) JobVO {
	var lastRunTime time.Time
	if job.Status.StartTime != nil {
		lastRunTime = job.Status.StartTime.Time
	}

	return JobVO{
		BaseVO: BaseVO{
			Name:      job.Name,
			Namespace: job.Namespace,
		},
		Alias:       job.Annotations["alias"],
		Describe:    job.Annotations["describe"],
		CreateTime:  job.CreationTimestamp.Time,
		LastRunTime: lastRunTime,
	}
}
